src/System.Private.CoreLib/shared/System/Threading/Tasks/Task.cs

```diff
@@ -700,6 +704,97 @@ private string DebuggerDisplayMethodDescription
+            if (etwLog.TasksSetActivityIds)
+                EventSource.SetCurrentThreadActivityId(TplEtwProvider.CreateGuidForTaskID(this.Id), out savedActivityID);
+            // previousTask holds the actual "current task" we want to report in the event
+            if (previousTask != null)
+                etwLog.TaskStarted(previousTask.m_taskScheduler.Id, previousTask.Id, this.Id);
+            else
+                etwLog.TaskStarted(TaskScheduler.Current.Id, 0, this.Id);
+        }
+
+        void EtwTaskCompleted(Task previousTask, ref Guid savedActivityID)
+        {
+            TplEtwProvider etwLog = TplEtwProvider.Log;
+            // previousTask holds the actual "current task" we want to report in the event
+            if (previousTask != null)
+                etwLog.TaskCompleted(previousTask.m_taskScheduler.Id, previousTask.Id, this.Id, IsFaulted);
+            else
+                etwLog.TaskCompleted(TaskScheduler.Current.Id, 0, this.Id, IsFaulted);
+
+            if (etwLog.TasksSetActivityIds)
+                EventSource.SetCurrentThreadActivityId(savedActivityID);
+        }
+
+        void EtwTaskWaitBegin()
+        {
+            TplEtwProvider etwLog = TplEtwProvider.Log;
+            Task currentTask = Task.InternalCurrent;
+            etwLog.TaskWaitBegin(
+                (currentTask != null ? currentTask.m_taskScheduler.Id : TaskScheduler.Default.Id), (currentTask != null ? currentTask.Id : 0),
+                this.Id, TplEtwProvider.TaskWaitBehavior.Synchronous, 0);
+        }
+
+        void EtwTaskWaitEnd()
+        {
+            TplEtwProvider etwLog = TplEtwProvider.Log;
+            Task currentTask = Task.InternalCurrent;
+            if (currentTask != null)
+            {
+                etwLog.TaskWaitEnd(currentTask.m_taskScheduler.Id, currentTask.Id, this.Id);
+            }
+            else
+            {
+                etwLog.TaskWaitEnd(TaskScheduler.Default.Id, 0, this.Id);
+            }
+            // logically the continuation is empty so we immediately fire
+            etwLog.TaskWaitContinuationComplete(this.Id);
+        }
+
+        internal void EtwTaskScheduled(TaskScheduler ts)
+        {
+            if ((m_stateFlags & Task.TASK_STATE_TASKSCHEDULED_WAS_FIRED) == 0)
+            {
+                m_stateFlags |= Task.TASK_STATE_TASKSCHEDULED_WAS_FIRED;
+
+                TplEtwProvider etwLog = TplEtwProvider.Log;
+
+                Task currentTask = Task.InternalCurrent;
+                Task parentTask = m_contingentProperties?.m_parent;
+                etwLog.TaskScheduled(ts.Id, currentTask == null ? 0 : currentTask.Id,
+                                     this.Id, parentTask == null ? 0 : parentTask.Id, (int)this.Options);
+            }
+        }
+
+        void EtwRunningContinuation(object continuationObject)
+        {
+            TplEtwProvider etw = TplEtwProvider.Log;
+            etw.RunningContinuation(Id, continuationObject);
```

Comment:
Same as below... the local etw isn't needed... and when we do have a local, it'd be nice to name it consistently, e.g. etw vs etwLog